src/graph/graphiti_extractor.py

```
@@ -286,6 +310,80 @@ def _build_entity_types(self) -> Dict[str, Type[BaseModel]]:
+        Check if chunk is purely structural (TOC, headers, page numbers).
+
+        These chunks have no semantic value for KG extraction.
+
+        Args:
+            content: Chunk text content
+
+        Returns:
+            True if chunk should be skipped
+        """
+        # Mostly numbers, dots and whitespace (page numbers, TOC numbers)
+        if re.match(r"^[\d\.\s\-–—]+$", content):
```

Ruff (0.14.7)

String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)